services/blue/src/api/Handlers/RoomTypeSelectionHandler.cs

```diff
@@ -22,23 +22,18 @@ public async Task Handle(SelectRoomType message, IMessageHandlerContext context)
         {
             await using var connection = await _connectionFactory.CreateAsync();
 
-            var query = $@"
-SELECT * FROM {RoomTypes}
-WHERE Id = @RoomTypeId AND Id NOT IN (
-    SELECT DISTINCT RoomTypeId FROM {Reservations}
-    WHERE Start >= @Start AND Start <= @End)";
+            var exists = await ExistsAsync(connection, message.RoomTypeId);
 
-            var queryResult = await connection.ExecuteQueryAsync<RoomType>(query, new
-            {
+            if (!exists)
+                throw new Exception();
+
+            var isAvailable = await IsAvailableAsync(
+                connection,
                 message.RoomTypeId,
                 message.Start,
-                message.End
-            });
-
-            var availableRoomType = queryResult.FirstOrDefault();
+                message.End);
 
-            if (availableRoomType is null)
-                // TODO: reply to caller with "fault" message.
+            if (isAvailable)
```

Comment:
Shouldn't that be when it is not available? 
```suggestion
            if (!isAvailable)
```

Reply:
yes :)